### DISCUSSION TITLE
detekt-cli-1.20.0-all.jar does not contain formatting plugin

### DISCUSSION THREAD

**Discussion by MarcelBochtler** [detekt/detekt#4790] [Q&A] [answered]
I'm using the CLI's `all` variant to run as a post-commit hook.
This is the `detekt.yml` I use: https://github.com/oss-review-toolkit/ort/blob/main/.detekt.yml

Using 1.19.0 this worked without any issues, however 1.20.0 seems to not include the formatting plugin anymore, and I get this error:
```
Property 'formatting' is misspelled or does not exist.
Run failed with 1 invalid config property.
	- Property 'formatting' is misspelled or does not exist.
```

If I add the formatting plugin to the executed command, the property `formatting` is allowed again. Is this expected?

**Comment by BraisGabin** [comment, marked as answer]
🤔 If the `cli` bounded at any time `formatting` directly that was an issue. `formatting` is a first-party plugin, but it's a plugin. It is not meant to be bounded inside the `cli`.

We had an issue that we were adding the configuration of `formatting` on our default configuration and that was fixed on 1.20.0 but it doesn't seem to be related with the issue that you are reporting.

If you want to enable `formatting` on the `cli` you need to add `-p path/to/the/formatting/jar`.

**Reply by bric3** [reply]
I ran to the same issue, and was thinking this plugin would be automatically resolved when `formatting` is encountered.

Also if the plugin has to be explicitly allowed on the command line, what about downloading automatically first-party plugins if specified with a name or an identifier, like `-p formatting` ?
